### DISCUSSION TITLE
Replace networkx with rustworkx

### DISCUSSION THREAD

**Discussion by thomasaarholt** [dbt-labs/dbt-core#9504] [Ideas]
Large (1000+ models) dbt projects can take quite a long time to build.
Part of that time is assumedly spent constructing a complex DAG of all the models in the project.

I see that [networkx is used in dbt,](https://github.com/dbt-labs/dbt-core/blob/2d59a51874672e5a95ec7f2f2aa18b8a66e9e738/core/dbt/compilation.py#L219) assumedly for this type of graph construction? If a significant amount of time is spent on this part, then it might be worth considering switching to the "spiritual successor" of networkx, rustworkx. It is significantly faster, as shown in these benchmarks: https://www.rustworkx.org/benchmarks.html

It isn't a _drop-in_ replacement, but it is not too far from it.

**Comment by yu-iskw** [comment]
I'm for this.
